### ISSUE TITLE
Unable to upload file to File upload field inside Repeater

### ISSUE THREAD

**Issue by aodihis** [verbb/formie#1419]
### Describe the bug

I have an issue with the file upload inside repeater. When I submit to  repeater field with more than 1 row and inside it the repeater I put the file upload the saved submission gone wrong.

The last row file upload is uploaded but it will save to the first row and the other file is lost.

### Steps to reproduce

1.  Create a repeater with file upload inside it
2. Submit the form here is the example data I sent.
 ```
 'repeater' : {
      'rows' :  [{ fielduploadField: "dataxxxx"}, { fielduploadField: "dataxxxx"}, { fielduploadField: "dataxxxx"}]
}
```
3. Check submission data. 

The all 3 repeated should have different file upload, but only the first row will have and it also wrong file that it should be for the third row.

### Form settings

- Multi-page form: Yes 
- Submission Method: GQL

### Craft CMS version

4.4.5

### Plugin version

2.0.28

### Multi-site?

No

**Comment by engram-design** [comment]
Seems to be GraphQL-specific, as normal submissions seem to work fine.

**Comment by jamesedmonston** [comment]
@engram-design I can confirm that it's isolated to forms being submitted through GraphQL. I've attached some screenshots of my form, the resulting submission, and the mutation being used.

It seems to be that it _always_ populates the first asset field in the repeaters.

<img width="1454" alt="Screenshot 2023-09-15 at 15 53 57" src="https://github.com/verbb/formie/assets/4851924/152f78d4-78fa-41b8-a050-81d46e61f324">
<img width="1455" alt="Screenshot 2023-09-15 at 15 54 52" src="https://github.com/verbb/formie/assets/4851924/6987d943-6557-4d27-8fd4-28c50b5151bd">
<img width="785" alt="Screenshot 2023-09-15 at 15 55 03" src="https://github.com/verbb/formie/assets/4851924/6c287191-ec96-446f-ad7d-6e04473dbdb2">

**Comment by engram-design** [comment]
Yep, I can replicate this but can't immediately figure out where this is occurring. Looking into it.

**Comment by alexjcollins** [comment]
Thanks, Josh. We really appreciate it.

**Comment by engram-design** [comment]
Managed to figure this one out. It's mostly due to Craft behaviour with Asset fields and using data values. The File Upload field inherits from Craft's Assets field. 

Firstly, Craft doesn't support multiple assets being created with data values. Secondly, there's a lack of handling for nested fields when using data values, and it assumes it's flat. I'm not even sure if it works in Matrix scenarios (which the Repeater field is essentially), as I couldn't get Matrix + Assets to work with mutations for testing.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.36"`.

**Comment by alexjcollins** [comment]
@engram-design Thanks for sorting this, I can confirm the changes have resolved the issue for us.

**Comment by engram-design** [comment]
Fixed in [2.0.37](https://github.com/verbb/formie/releases/tag/2.0.37)
